### PR TITLE
Put under flag nginx ingress syncer and make wordpress domains optional

### PR DESCRIPTION
**Pull request by AMecea**

### CODE REVIEW

File: pkg/controller/wordpress/wordpress_controller_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ package wordpress
 
 import (
 	"fmt"
+	"github.com/presslabs/wordpress-operator/pkg/internal/wordpress"
```
Comment (vtemian) [inline pkg/controller/wordpress/wordpress_controller_test.go]: Wrong imports order.

File: pkg/apis/wordpress/v1alpha1/wordpress_types.go [hunk 2]
```diff
@@ -54,8 +54,7 @@ type WordpressSpec struct {
 	Replicas *int32 `json:"replicas,omitempty"`
 	// Domains for which this this site answers.
 	// The first item is set as the "main domain" (eg. WP_HOME and WP_SITEURL constants).
-	// +kubebuilder:validation:MinItems=1
-	Domains []Domain `json:"domains"`
+	Domains []Domain `json:"domains,omitempty"`
```
Comment (vtemian) [inline pkg/apis/wordpress/v1alpha1/wordpress_types.go]: Why do we let creation without at least a domain?

File: pkg/internal/wordpress/pod_template.go [hunk 3]
```diff
@@ -339,7 +339,7 @@ func (wp *Wordpress) installWPContainer() []corev1.Container {
 	if len(wp.Spec.TLSSecretRef) > 0 {
 		scheme = "https"
 	}
-	url := fmt.Sprintf("%s://%s/", scheme, wp.Spec.Domains[0])
+	url := fmt.Sprintf("%s://%s/", scheme, wp.MainDomain())
```
Comment (vtemian) [inline pkg/internal/wordpress/pod_template.go]: We can have a method called `wp.URL()`

File: pkg/apis/wordpress/v1alpha1/wordpress_types.go (hunk 2 above)
Comment (AMecea) [reply pkg/apis/wordpress/v1alpha1/wordpress_types.go]: This is what we agreed. The default domain is `<wp name>.<namespace>.svc.cluster.local` if not specified.

File: pkg/internal/wordpress/pod_template.go (hunk 3 above)
Comment (AMecea) [reply pkg/internal/wordpress/pod_template.go]: I don't think it's necessary.

Comment (vtemian) [reply pkg/internal/wordpress/pod_template.go]: Just for the convenience and since we're re-using it a lot, I guess we can do it.

Comment (AMecea) [reply pkg/internal/wordpress/pod_template.go]: Done

File: pkg/internal/wordpress/wordpress.go [hunk 4]
```diff
@@ -126,3 +126,24 @@ func (o *Wordpress) JobPodLabels() labels.Set {
...
+
+// HomeURL returns the WP_HOMEURL (e.g. http://example.com/ )
+func (o *Wordpress) HomeURL() string {
+
+	scheme := "http"
+	if len(o.Spec.TLSSecretRef) > 0 {
+		scheme = "https"
+	}
+
+	return fmt.Sprintf("%s://%s/", scheme, o.MainDomain())
```
Comment (vtemian) [inline pkg/internal/wordpress/wordpress.go]: You don't the last `/`.

File: pkg/internal/wordpress/wordpress.go [hunk 5]
```diff
@@ -126,3 +126,24 @@ func (o *Wordpress) JobPodLabels() labels.Set {
...
+	if len(o.Spec.Domains) > 0 {
+		return o.Spec.Domains[0]
+	}
+
+	// return the local cluster name that points to wordpress service
+	return wordpressv1alpha1.Domain(fmt.Sprintf("%s.%s.svc.cluster.local", o.ComponentName(WordpressService), o.Namespace))
+}
+
+// HomeURL returns the WP_HOMEURL (e.g. http://example.com/ )
+func (o *Wordpress) HomeURL() string {
```
Comment (vtemian) [inline pkg/internal/wordpress/wordpress.go]: Let's add some unit tests for it.

File: pkg/internal/wordpress/wordpress.go [hunk 6]
```diff
@@ -126,3 +126,24 @@ func (o *Wordpress) JobPodLabels() labels.Set {
 	l["app.kubernetes.io/component"] = "wp-cli"
 	return l
 }
+
+// MainDomain returns the site main domain or a local domain <cluster-name>.<namespace>.svc.cluster.local
+func (o *Wordpress) MainDomain() wordpressv1alpha1.Domain {
```
Comment (vtemian) [inline pkg/internal/wordpress/wordpress.go]: Let's add some unit tests for it.

File: pkg/internal/wordpress/wordpress.go (hunk 5 above)
Comment (AMecea) [reply pkg/internal/wordpress/wordpress.go]: Done

File: pkg/internal/wordpress/wordpress.go (hunk 6 above)
Comment (AMecea) [reply pkg/internal/wordpress/wordpress.go]: Done

**Comment by calind** [comment]
Closed in favor of #51.

**Pull request closed without merging**